openhgnn/models/MAGNN.py

```diff
@@ -433,56 +476,98 @@ def mp_instance_sampler(g, metapath_list):
+                                         how='inner')
+                if col_i != col_j:
+                    res[metapath].drop(columns=col_j, inplace=True)
+            res[metapath] = res[metapath].values
+
+        with open(file_addr, 'wb') as file:
+            pickle.dump(res, file)
 
     return res
 
-def svm_test(X, y, test_sizes=(0.2, 0.4, 0.6, 0.8), repeat=10):
-    # This method is implemented by author
-    random_states = [182318 + i for i in range(repeat)]
-    result_macro_f1_list = []
-    result_micro_f1_list = []
-    for test_size in test_sizes:
-        macro_f1_list = []
-        micro_f1_list = []
-        for i in range(repeat):
-            X_train, X_test, y_train, y_test = train_test_split(
-                X, y, test_size=test_size, shuffle=True, random_state=random_states[i])
-            svm = LinearSVC(dual=False)
-            svm.fit(X_train, y_train)
-            y_pred = svm.predict(X_test)
-            macro_f1 = f1_score(y_test, y_pred, average='macro')
-            micro_f1 = f1_score(y_test, y_pred, average='micro')
-            macro_f1_list.append(macro_f1)
-            micro_f1_list.append(micro_f1)
-        result_macro_f1_list.append((np.mean(macro_f1_list), np.std(macro_f1_list)))
-        result_micro_f1_list.append((np.mean(micro_f1_list), np.std(micro_f1_list)))
-    return result_macro_f1_list, result_micro_f1_list
+def mini_mp_instance_sampler(seed_nodes, mp_instances, num_samples):
+    mini_mp_inst = {}
+    metapath_list = list(mp_instances.keys())
+
+    for ntype in seed_nodes.keys():
+        target_mp_types = np.array(metapath_list)[[metapath[0] == ntype for metapath in metapath_list]]
+        for metapath in target_mp_types:  # the metapath instances of the certain metapath
+            for node in seed_nodes[ntype]:
+                # _mp_inst = np.isin(mp_instances[metapath][:, 0], seed_nodes[ntype])
+                _mp_inst = mp_instances[metapath][mp_instances[metapath][:, 0] == node]
+                # _mp_inst = mp_instances[metapath][_mp_inst]
+                dst_nodes, dst_counts = np.unique(_mp_inst[:, -1], return_counts=True)
+
+                # the method of computing sampling probabilities originates from author's codes
+                p = []
+                for count in dst_counts:
+                    p += [(count ** (3 / 4)) / count] * count
+                p = np.array(p)
+                p = p / p.sum()
+
+                _num_samples = min(num_samples, len(p))
+                mp_choice = np.random.choice(len(p), _num_samples, replace=False, p=p)
+                if metapath not in mini_mp_inst.keys():
+                    mini_mp_inst[metapath] = _mp_inst[mp_choice]
+                else:
+                    mini_mp_inst[metapath] = np.concatenate((mini_mp_inst[metapath], _mp_inst[mp_choice]),
+                                                            axis=0)
+
+    return mini_mp_inst
+
+# def svm_test(X, y, test_sizes=(0.2, 0.4, 0.6, 0.8), repeat=10):
```

Comment:
Remove if it's no longer used?

Reply:
Actually we still need this method. We're considering about putting this method in a trainerflow of MAGNN, which I'll implement later. I'll delete this method after implementing the trainerflow.

Reply:
K.